demos/animation/anim.go

```diff
@@ -61,18 +61,18 @@ func (game *GameWorld) CreateEntity(point *engi.Point, spriteSheet *engi.Sprites
 }
 
 type ControlSystem struct {
-	*ecs.System
+	*ecs.BasicSystemWrapper
 }
 
-func (ControlSystem) Type() string {
-	return "ControlSystem"
-}
+func (*ControlSystem) Type() string { return "ControlSystem" }
+func (*ControlSystem) Pre()         {}
+func (*ControlSystem) Post()        {}
 
 func (c *ControlSystem) New(*ecs.World) {
-	c.System = ecs.NewSystem()
+	c.BasicSystemWrapper = ecs.NewBasicSystemWrapper(c)
 }
 
-func (c *ControlSystem) Update(entity *ecs.Entity, dt float32) {
+func (c *ControlSystem) EUpdate(entity *ecs.Entity, dt float32) {
```

Comment:
The difference is that the Update function takes only a time difference float as parameter, while EUpdate (E for Entity) takes both an Entity and the float. 

On December 31, 2015 11:15:30 PM GMT+01:00, Harrison Shoebridge <notifications@github.com> wrote:
>>  }
>>  
>> -func (c *ControlSystem) Update(entity *ecs.Entity, dt float32) {
>> +func (c *ControlSystem) EUpdate(entity *ecs.Entity, dt float32) {
>
>Yeah, I don't like this function name.
>
>Also: What is the difference here? The function signatures look
>identical, I don't think people could tell just be looking.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/paked/engi/pull/138/files#r48672646

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.